common/beerocks/bcl/source/beerocks_string_utils.cpp

```diff
@@ -114,7 +114,6 @@ std::string string_utils::int_to_hex_string(const unsigned int integer,
     // for example: int_to_hex_string(255, 4) -> "00ff"
     //              int_to_hex_string(255, 1) -> "ff"
 
-    std::string return_string;
```

Comment:
I really liked your first few commits: they are really clear and though they are large, it's trivial to check that they do the right thing.

Here, however, it's a lot more difficult, because there are two different kind of changes in the same commit, and the commit message doesn't describe them. So I'd like this to be split into two commits: the first one that does the string replacements, and the second one that fixes this issue (which I assume is an unused variable).